salt/runners/jobs.py

```diff
@@ -305,68 +305,83 @@ def list_jobs(ext_source=None,
         )
     mminion = salt.minion.MasterMinion(__opts__)
 
-    ret = mminion.returners['{0}.get_jids'.format(returner)]()
-
-    mret = {}
-    for item in ret:
-        _match = True
-        if search_metadata:
-            _match = False
-            if 'Metadata' in ret[item]:
-                if isinstance(search_metadata, dict):
-                    for key in search_metadata:
-                        if key in ret[item]['Metadata']:
-                            if ret[item]['Metadata'][key] == search_metadata[key]:
+    if __opts__.get('filter') == 'native':
+        fun = '{0}.get_jids_native'.format(returner)
+        if fun not in mminion.returners:
```

Comment:
If we have this config value set then this will prevent the returner from running at all even if this function is not in there

Reply:
Yes, I just copy what was done for get_jids_filter